src/shared/extensions/yaml.ts

```diff
@@ -32,28 +32,67 @@ function evaluate(schema: vscode.Uri | (() => vscode.Uri)): vscode.Uri {
 }
 
 export interface YamlExtension {
-    assignSchema(path: vscode.Uri, schema: vscode.Uri | (() => vscode.Uri)): void
-    removeSchema(path: vscode.Uri): void
-    getSchema(path: vscode.Uri): vscode.Uri | undefined
+    assignSchema(path: string, registry: string | undefined, schema: vscode.Uri | (() => vscode.Uri)): void
+    removeSchema(path: string, registry: string | undefined): void
+    getSchema(path: string): string | undefined
+}
+
+function registerSchema(
+    path: string,
+    registry: string | undefined,
+    schema: vscode.Uri | (() => vscode.Uri),
+    schemaMap: Map<string, Map<string, vscode.Uri>>
+) {
+    if (!registry) {
+        return
+    }
+    const schemas = schemaMap.get(path)
+    if (!schemas) {
+        schemaMap.set(path, new Map())
+    }
+    schemaMap.set(path, schemaMap.get(path)!.set(registry, applyScheme(AWS_SCHEME, evaluate(schema))))
+}
+
+function deRegisterSchema(path: string, registry: string | undefined, schemaMap: Map<string, Map<string, vscode.Uri>>) {
```

Comment:
"unregister" is more conventional/common. This is a nit, but also a general pattern: use common, simple names unless there is an explicit reason to use an uncommon name.

```suggestion
function unregisterSchema(path: string, registry: string | undefined, schemaMap: Map<string, Map<string, vscode.Uri>>) {
```